### PR TITLE
chore: Updating List item and List components in ADS to start using it in Entity explorer

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Under Entity explorer modularisation, we are updating the List item and List components in ADS which will eventually be getting used in Entity explorer to show the entity items. This change makes sure current usages of this component doesn't break.

Fixes [#37620](https://github.com/appsmithorg/appsmith/issues/37620)

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12517459729>
> Commit: e1601a6a461cb860787a62edfb6273a94bb086bd
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12517459729&attempt=2&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.All
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Sanity/Datasources/MockDBs_Spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Fri, 27 Dec 2024 17:09:22 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a flexible `rightControl` property for list items, enhancing interactivity.
	- Added `rightControlVisibility` property to manage control visibility based on hover state.
	- Implemented a new `customTitleComponent` property for additional customization options.

- **Bug Fixes**
	- Adjusted rendering logic for descriptions in list items to improve clarity.

- **Style**
	- Enhanced styling for list items, including new padding and border-radius adjustments.
	- Updated CSS selectors for improved specificity in list item styling.

- **Documentation**
	- Updated component stories to reflect new properties and improved API clarity.

### CODE REVIEW

File: app/client/packages/design-system/ads/src/List/List.styles.tsx [hunk 1]
```diff
@@ -36,30 +36,63 @@ export const StyledList = styled.div`
   gap: var(--ads-v2-spaces-2);
 `;
 
-export const Wrapper = styled.div`
+export const StyledListItem = styled.div<{
+  isBlockDescription?: boolean;
+  rightControlVisibility?: "hover" | "always";
+  size: ListSizes;
```
Comment (hetunandu) [inline app/client/packages/design-system/ads/src/List/List.styles.tsx:42]: Can we pass these states via [data-{xyz}] properties instead? Passing this as arguments causes a lot of style rerenders

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/packages/design-system/ads/src/List/List.styles.tsx (hunk 1 above)
Comment (ankitakinger) [reply app/client/packages/design-system/ads/src/List/List.styles.tsx:42]: I don't think it will be possible for `size`. Can do for others.

**Review by hetunandu — DISMISSED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**